### ISSUE TITLE
Provide scaleXTimeOffset for relative duration axis

### ISSUE THREAD

**Issue by vlsi** [JetBrains/lets-plot#468]
The use case is to show "time since test start" axis.

The axis labels should be formatted like "2h 30m", "1d 2h", "1w 2d 3h", or something like that.

**Comment by IKupriyanov-HORIS** [comment]
Hi! Could you say what kind of data is in input? Is it UNIX-time, formatted strings, zero-based nanosecond events?

**Comment by vlsi** [comment]
Here's the use-case: https://github.com/apache/jmeter/pull/674#issuecomment-955780938

The chart displays "arrival rate" vs "time since test started" data.
I prepare the data, so I can prepare it in whatever format is needed.

I think zero-based (nano)second events would work the best, since the display of UNIX-time requires the notion of "timezone", and I do not have timezones (the meaning of the axis is "time since the beginning of the test")

**Comment by vlsi** [comment]
Just in case, for now, I use a numeric horizontal axis, and I perform manual scaling which adjusts units depending on the total duration. It turned out to be not that bad to show the horizontal axis as "numeric hours" or "numeric minutes".

```kotlin
        val totalDuration = time.last()
        val timeUnit = when {
            totalDuration > 2.5 * TimeUnit.DAYS.toSeconds(1) -> TimeUnit.DAYS
            totalDuration > 2.5 * TimeUnit.HOURS.toSeconds(1) -> TimeUnit.HOURS
            totalDuration > 2.5 * TimeUnit.MINUTES.toSeconds(1) -> TimeUnit.MINUTES
            else -> TimeUnit.SECONDS
        }
        if (timeUnit != TimeUnit.SECONDS) {
            val scale = 1.0 / timeUnit.toSeconds(1)
            for (i in time.indices) {
                time[i] *= scale
            }
        }
```

**Comment by IKupriyanov-HORIS** [comment]
Thank you! Will check how we can implement this.

**Comment by IKupriyanov-HORIS** [comment]
Thank you for your feedback! `scale_x_time` and `scale_y_time` are implemented in [v2.1.1](https://github.com/JetBrains/lets-plot/releases/tag/v2.2.1). Make us know if there are other improvements can be made.
